### PR TITLE
[GraphBolt] Store `ORIGINAL_EDGE_ID` in CSCSamplingGraph's `edge_attributes`.

### PR DESCRIPTION
**Pull request by keli-wen**
## Description

I've made modification to `gb.from_dglgraph` function to store the `ORIGINAL_EDGE_ID` within the CSCSamplingGraph.

Additionally, I've implemented the requisite unit test to validate the correctness of the `ORIGINAL_EDGE_ID` mapping. By utilizing the `CSCSamplingGraph` and the mapping, we can retrieve node pairs from the `DGLGraph` and verify their equalities. 

More Specifically:

### For HomoGraph Test
```mermaid
graph TD;
    Start(Start) --> GetCOO[Generate COO Representation];
    GetCOO --> CheckEdgeMapping[Verify Edge Mapping using ORIGINAL_EDGE_ID];
    CheckEdgeMapping --> End(End);
```

### For HeteroGraph Test
```mermaid
graph TD;
    Start(Start) --> MapNodeIDs["Create Node ID Mapping (GB -> DGL)"];
    MapNodeIDs --> GetCOO[Generate COO Representation by Node ID Mapping];
    GetCOO --> VerifyEtypes[Verify GraphBolt and DGL Edge Type Order];
    VerifyEtypes --> CheckEdgeMapping[Validate Edge Mapping using ORIGINAL_EDGE_ID];
    CheckEdgeMapping --> End(End);
```

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [x] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/impl/csc_sampling_graph.py [hunk 1]
```diff
@@ -760,13 +760,16 @@ def from_dglgraph(g: DGLGraph, is_homogeneous=False) -> CSCSamplingGraph:
     # Assign edge type according to the order of CSC matrix.
     type_per_edge = None if is_homogeneous else homo_g.edata[ETYPE][edge_ids]
 
+    # Assign edge attributes according to the original eids mapping.
+    edge_attributes = {ORIGINAL_EDGE_ID: homo_g.edata[EID][edge_ids]}
```
Comment (frozenbugs) [inline python/dgl/graphbolt/impl/csc_sampling_graph.py:764]: Should we add option in the from_dglgraph and default it to false, such that if no eid needed in the task, we can avoid the memory cost for storing the eid,

File: tests/python/pytorch/graphbolt/impl/test_csc_sampling_graph.py [hunk 2]
```diff
@@ -1218,6 +1218,22 @@ def test_from_dglgraph_homogeneous():
     dgl_g = dgl.rand_graph(1000, 10 * 1000)
     gb_g = gb.from_dglgraph(dgl_g, is_homogeneous=True)
 
+    # Get the COO representation of the CSCSamplingGraph.
+    gb_coo = [
```
Comment (peizhou001) [inline tests/python/pytorch/graphbolt/impl/test_csc_sampling_graph.py:1222]: Another choice to get column is
```
num = indptr[1:] - indptr[:-1]
column = torch.arange(indptr.size(0))
column = column.repeat_interleave(num)
```

you can choose the one you like

File: tests/python/pytorch/graphbolt/impl/test_csc_sampling_graph.py [hunk 3]
```diff
@@ -1218,6 +1218,22 @@ def test_from_dglgraph_homogeneous():
...
+        # row -> col.
+        (gb_g.indices[row_idx], torch.tensor(col))
+        for col in range(gb_g.total_num_nodes)
+        for row_idx in range(gb_g.csc_indptr[col], gb_g.csc_indptr[col + 1])
+    ]
+
+    # Use ORIGINAL_EDGE_ID to check if the edge mapping is correct.
+    for edge_idx in range(gb_g.total_num_edges):
```
Comment (peizhou001) [inline tests/python/pytorch/graphbolt/impl/test_csc_sampling_graph.py:1230]: You can also use tensor operation here.

File: tests/python/pytorch/graphbolt/impl/test_csc_sampling_graph.py [hunk 4]
```diff
@@ -1250,6 +1266,46 @@ def test_from_dglgraph_heterogeneous():
...
+    # `reverse_node_id` is used to map the node id in CSCSamplingGraph to the
+    # node id in Hetero-DGLGraph.
+    reverse_node_id = {}
+    num_ntypes = len(gb_g.node_type_offset) - 1
+    for i in range(0, num_ntypes):
+        for j in range(gb_g.node_type_offset[i], gb_g.node_type_offset[i + 1]):
+            reverse_node_id[j] = torch.tensor(j) - gb_g.node_type_offset[i]
+
+    # Get the COO representation of the CSCSamplingGraph.
+    gb_coo = [
```
Comment (peizhou001) [inline tests/python/pytorch/graphbolt/impl/test_csc_sampling_graph.py:1278]: Ditto

File: tests/python/pytorch/graphbolt/impl/test_csc_sampling_graph.py [hunk 5]
```diff
@@ -1250,6 +1266,46 @@ def test_from_dglgraph_heterogeneous():
...
+            map(
+                lambda ss: ss.split(":")[1],
+                gb_g.metadata.edge_type_to_id.keys(),
+            )
+        )
+        == dgl_g.etypes
+    )
+
+    # Use ORIGINAL_EDGE_ID to check if the edge mapping is correct.
+    for edge_idx in range(gb_g.total_num_edges):
```
Comment (peizhou001) [inline tests/python/pytorch/graphbolt/impl/test_csc_sampling_graph.py:1299]: Ditto

File: python/dgl/graphbolt/impl/csc_sampling_graph.py (hunk 1 above)
Comment (keli-wen) [reply python/dgl/graphbolt/impl/csc_sampling_graph.py:764]: https://github.com/dmlc/dgl/issues/6406 issue filed

File: tests/python/pytorch/graphbolt/impl/test_csc_sampling_graph.py (hunk 2 above)
Comment (keli-wen) [reply tests/python/pytorch/graphbolt/impl/test_csc_sampling_graph.py]: Done

**Review by peizhou001 — APPROVED** [review]

**Pull request merged**